etc/notebooks/got_scotch_demo/scotch_dashboard_declarative.ipynb

```diff
@@ -0,0 +1,783 @@
+     }
+    }
+   },
+   "source": [
+    "### Feature Chart"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "Lastly, we will display the scotches features with a variety of urth-viz-* elements.\n",
+    "\n",
+    "We need to use urth-core-function to call *get_feature_keys*, give it the selected scotch name, the similar scotch name user clicked on, and the result of the function call is a list of these two names. We will assign the result to a variable called *feature_keys* that another function *get_features_for_chart* will make use of right after this."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "%%html\n",
+    "<template is=\"urth-core-bind\">\n",
+    "    <urth-core-function id=\"getFeatureKeys\" ref=\"get_feature_keys\" arg-scotch=\"{{selectedScotch}}\" arg-similar=\"{{similarScotch}}\" result=\"{{feature_keys}}\" auto></urth-core-function>\n",
+    "</template>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "Similarly to the suggestion table, we need to call function *get_features_for_chart* to retrieve the data for urth-viz-* elements.\n",
+    "\n",
+    "Once the user clicks one of the 5 similar scotches from the suggestion table above, *feature_keys* variable is updated to include the clicked scotch, and urth-core-function will call its function again to get new feature table, and urth-viz-* will re-render to use this newly updated feature table."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "%%html\n",
+    "<template is=\"urth-core-bind\">\n",
+    "    <urth-core-function id=\"getFeaturesFunc\" ref=\"get_features_for_chart\" arg-feature_keys=\"{{feature_keys}}\" result=\"{{featuresTbl}}\" auto></urth-core-function>\n",
+    "</template>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "Now, we define a template to allow user switch between different types of chart. We start with setting up an iron-selector element which includes the 3 icons for table, bar and pie chart. When user clicks on any of those 3 icons, the name of the chart is stored as variable *chartType*. Then we set up iron-pages with 3 children pages that contain the three types of chart, each chart type per child page. This iron-pages element will depend on what the variable *chartType* is and show/hide the proper child page."
```

Comment:
I'm totally lost on what the pie chart is showing me. I expect a (violent) negative reaction from folks attuned to visualization about it too. Let's just stick with bar and table if those are the two that function properly and are reasonable for comparing properties of two items.